test/operators/second_quantization/test_fermionic_op.py

```diff
@@ -18,73 +18,102 @@
 
-from qiskit_nature import QiskitNatureError
 from qiskit_nature.operators import FermionicOp
 
 
 def fermion_labels(length):
     """Generate list of fermion labels with given length."""
-    return [
-        "".join(label) for label in product(["I", "+", "-", "N", "E"], repeat=length)
-    ]
+    return ["".join(label) for label in product(["I", "+", "-", "N", "E"], repeat=length)]
 
 
 @ddt
 class TestFermionicOp(QiskitNatureTestCase):
     """FermionicOp tests."""
 
+    def assertFermionEqual(self, first: FermionicOp, second: FermionicOp):
+        """Fail if two FermionicOps have different.
+        Note that this equality check is approximated since the true equality check is costly.
+        """
+        self.assertSetEqual(frozenset(first.to_list()), frozenset(second.to_list()))
+
     @data(*fermion_labels(1), *fermion_labels(2))
     def test_init(self, label):
         """Test __init__"""
         self.assertListEqual(FermionicOp(label).to_list(), [(label, 1)])
 
-    def test_init_invalid_label(self):
+    @data("INX", [("++", 1), ("EF", 1)], "", "+_0 -_2")
+    def test_init_invalid_label(self, labels):
         """Test __init__ with invalid label"""
-        with self.assertRaises(QiskitNatureError):
-            FermionicOp("INX")
-        with self.assertRaises(QiskitNatureError):
-            FermionicOp([("++", 1), ("EF", 1)])
+        with self.assertRaises(ValueError):
+            FermionicOp(labels)
 
     def test_init_multiterm(self):
         """Test __init__ with multi terms"""
         labels = [("N", 2), ("-", 3.14)]
         self.assertListEqual(FermionicOp(labels).to_list(), labels)
 
-    def test_init_invalid(self):
-        """Test invalid __init__"""
-        with self.assertRaises(QiskitNatureError):
-            FermionicOp("test")
+    @data(*fermion_labels(1))
+    def test_init_sparse_label(self, label):
+        """Test __init__ with sparse label"""
+        fer_op = FermionicOp(f"{label}_0", register_length=1)
+        targ = FermionicOp([(label, 1)])
+        self.assertFermionEqual(fer_op, targ)
+
+    @data(*fermion_labels(2))
+    def test_init_sparse_label_len_four(self, label):
+        """Test __init__ with sparse label"""
+        fer_op = FermionicOp(f"{label[0]}_3 {label[1]}_0", register_length=4)
+        targ = FermionicOp([(f"{label[0]}II{label[1]}", 1)])
+        self.assertFermionEqual(fer_op, targ)
+
+    def test_init_multiple_digits(self):
+        """Test __init__ for sparse label with multiple digits"""
+        actual = FermionicOp([("+_10 -_2", 1 + 2j), ("-_12", 56)], register_length=13)
+        desired = [
+            ("II+IIIIIII-II", 1 + 2j),
+            ("-IIIIIIIIIIII", 56),
+        ]
+        self.assertListEqual(actual.to_list(), desired)
 
     def test_neg(self):
         """Test __neg__"""
         fer_op = -FermionicOp("+N-EII")
-        self.assertListEqual(fer_op.to_list(), [("+N-EII", -1)])
+        targ = FermionicOp([("+N-EII", -1)])
```

Comment:
What does `targ` stand for?

Reply:
target. In terra, this is used (https://github.com/Qiskit/qiskit-terra/search?q=targ)